sw/host/opentitanlib/src/io/jtag.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+use std::rc::Rc;
+use structopt::StructOpt;
+use thiserror::Error;
+
+use crate::app::TransportWrapper;
+use crate::dif::lc_ctrl::{DifLcCtrlState, LcCtrlReg};
+use crate::impl_serializable_error;
+
+#[derive(Debug, StructOpt)]
+pub struct JtagParams {
+    #[structopt(long, help = "OpenOCD binary path.", default_value = "openocd")]
+    pub openocd: String,
+
+    #[structopt(long, help = "Path to OpenOCD JTAG adapter config file.")]
+    openocd_adapter_config: String,
+}
+
+impl JtagParams {
+    pub fn create(&self, transport: &TransportWrapper) -> Result<Rc<dyn Jtag>> {
+        let jtag = transport.jtag(&self.openocd, &self.openocd_adapter_config)?;
+        Ok(jtag)
+    }
+}
+
+/// Errors related to the JTAG interface.
+#[derive(Error, Debug, Deserialize, Serialize)]
+pub enum JtagError {
+    #[error("JTAG timeout")]
+    Timeout,
+    #[error("JTAG busy")]
+    Busy,
+    #[error("Generic error {0}")]
+    Generic(String),
+}
+impl_serializable_error!(JtagError);
+
+/// Represents an OpenOCD command sent over JTAG.
+pub enum OpenOcdCmd {
+    ReadLcCtrlReg(LcCtrlReg),
+    WriteLcCtrlReg(LcCtrlReg, u32),
+    // ReadCpuReg(IbexReg),
```

Comment:
This was commented b/c we need to generate/define an enum like the [LcCtrlReg enum](https://cs.opensource.google/opentitan/opentitan/+/master:sw/host/opentitanlib/src/dif/lc_ctrl.rs;drc=312e9ddeb49dedca0454a3bd7423e3bf86a51ca4;l=73) but for Ibex CSRs. This can be done in a follow up.